docs/installing-solidity.rst

```diff
@@ -6,6 +6,28 @@
 Installing Solidity
 ###################
 
+Versioning
+==========
+
+Solidity versions follow `semantic versioning <https://semver.org>` and in addition to
+releases, **nightly development builds** are also made available.  The nightly builds
+are not guaranteed to be working and despite best efforts they might contain undocumented
+and/or broken changes.
+
+After a release is made, the patch version level is bumped, because we assume that only
+patch level (non-breaking and non-feature) changes follow. When changes are merged, the
+version should be bumped according to semver and the severity of the change. Finally,
+a release is always made with the version of the current nightly build.
+
```

Comment:
Perhaps also add that nightly builds always have a prerelease specifier and thus are considered "older" than the actual release.